bacon/commons/Observer.swift

```diff
@@ -0,0 +1,21 @@
+//
+//  Observer.swift
+//  bacon
+//
+//  Created by Fabian Terh on 27/3/19.
+//  Copyright © 2019 nus.CS3217. All rights reserved.
+//
+
+import Foundation
+
+/// Observer protocol in the Observer-Subject pattern.
+/// - Note: This protocol only allows conformance by classes,
+/// as `unregisterObserver()` in Observable makes use of identity comparison.
+protocol Observer: class {
+
+    /// Receives notification of a value published by the observed subject.
+    /// Since `value` is typed as `Any`, the observer is responsible for casting `value` to
+    /// the correct type of whatever it is observing.
+    func notify(_ value: Any)
```

Comment:
should this be notified instead? notify sounds abit confusing haha

Reply:
I think notify is fine, all the observer patterns implementations I found use the imperative form of the word too haha